java/vector/src/main/java/org/apache/arrow/vector/ipc/message/ArrowRecordBatch.java

```diff
@@ -89,7 +107,9 @@ public ArrowRecordBatch(
     List<ArrowBuffer> arrowBuffers = new ArrayList<>(buffers.size());
     long offset = 0;
     for (ArrowBuf arrowBuf : buffers) {
-      arrowBuf.getReferenceManager().retain();
+      if (retainBuffers) {
```

Comment:
Could you try to explain why working with compression options impact or is needed to add this flag  "retainBuffers"? 

What will be the error if this flag retainBuffers is not added?

Reply:
https://github.com/apache/arrow/blob/8199fa79a4440ca3976c5712ccf3bc3eaff0eae2/java/vector/src/main/java/org/apache/arrow/vector/VectorUnloader.java#L102-L105